src/lib/comms/sol-http-common.c

```diff
@@ -843,15 +844,27 @@ sol_http_split_query(const char *query, struct sol_http_params *params)
         }
 
         if (!sol_http_param_add_copy(params,
-            CREATE_QUERY_PARAM(key, value))) {
+            CREATE_PARAM(key, value))) {
             SOL_ERR("Could not add the HTTP param %.*s:%.*s",
                 SOL_STR_SLICE_PRINT(key), SOL_STR_SLICE_PRINT(value));
             goto exit;
         }
     }
-#undef CREATE_QUERY_PARAM
+#undef CREATE_PARAM
 
 exit:
     sol_vector_clear(&tokens);
     return 0;
 }
+
+SOL_API int
+sol_http_split_query(const char *query, struct sol_http_params *params)
+{
+    return sol_http_split_str_key_value(query, SOL_HTTP_PARAM_QUERY_PARAM, params);
+}
```

Comment:
yes, it's possible but it's not only the delimiter, for headers, post fields and query params it's allowed empty values, but i guess it's not valid for cookies. Also, it'll make redundant the functions to split query and post fields, and still will be necessary functions for things that is not key/value.

Do you think it's worth export `sol_http_split_str_key_value()` and remove `sol_http_split_query()` and `sol_http_split_post_field()` ?